### ISSUE TITLE
Decide on a consistent usage of `pool` vs `type` in RPCs

### ISSUE THREAD

**Issue by str4d** [zcash/zcash#5534]
We use `pool` and `type` as keys in various places. We should try and be consistent about it.

**Comment by nuttycom** [comment]
* Note usage - what pool funds are in - use "pool" and deprecate any existing "type" field.
* Address type: p2pkh, p2sh, Sprout, Sapling, Unified
* Item types for unified addresses & keys.

**Comment by LarryRuane** [comment]
This will be closed by #5719 (which does more than its title suggests).

**Comment by LarryRuane** [comment]
From the description above:
> We use `pool` and `type` as keys in various places. We should try and be consistent about it.

I'm also unsure of the difference between `pool` and `receiver_type`. Are the definitions of those two (and address type) documented somewhere, or, if not, should we?

**Comment by steven-ecc** [comment]
[ZIP 316](https://zips.z.cash/zip-0316) is the ZIP for Unified Addresses and contains a pretty good writeup on pools, receivers, etc.

TL;DR - a pool and type can be synonymous in some cases (maybe most). A receiver `type` can be Orchard, Sapling, or transparent. When used in that context, type would be the correct term to use. Orchard, Sapling, and transparent can also refer to `pools` of course. 

Not sure if that helps or is more confusing. =)
